CSharp/BatchExplorer/Views/MainView.xaml

```diff
@@ -720,6 +744,13 @@
                                 </ContextMenu>
                             </DataGrid.ContextMenu>
                             <DataGrid.Columns>
+                                <DataGridTemplateColumn>
+                                    <DataGridTemplateColumn.CellTemplate>
+                                        <DataTemplate>
+                                            <CheckBox IsChecked="{Binding IsChecked, UpdateSourceTrigger=PropertyChanged}" />

```

Comment:
Why are you calling the UpdateSourceTrigger? The binding should happen auto-magically. Perhaps you need to set two-mode binding instead?